--- origsrc/lrzip-0.616/libzpaq/libzpaq.cpp	2013-02-07 06:22:06.000000000 -0600
+++ src/lrzip-0.616/libzpaq/libzpaq.cpp	2014-03-09 00:24:23.259527400 -0600
@@ -1844,7 +1844,7 @@ int ZPAQL::assemble() {
     put5(0x48897424,8);       // mov [rsp+8], rsi
     put5(0x48895424,16);      // mov [rsp+16], rdx
     put5(0x48894c24,24);      // mov [rsp+24], rcx
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__CYGWIN__)
     put2l(0x48bf, this);      // mov rdi, this
 #else  // Windows
     put2l(0x48b9, this);      // mov rcx, this
@@ -2385,7 +2385,7 @@ int Predictor::assemble_p() {
   if (S==4)
     put4(0x8b7c2414);         // mov edi,[esp+0x14] ; pr
   else {
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
     put3(0x4889cf);           // mov rdi, rcx (1st arg in Win64)
 #endif
   }
@@ -2775,7 +2775,7 @@ int Predictor::assemble_p() {
     put4(0x8b6c2418);          // mov ebp,[esp+0x18] ; (2nd arg = y)
   }
   else {
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__CYGWIN__)
     put3(0x4889f5);            // mov rbp, rsi (2nd arg in Linux-64)
 #else
     put3(0x4889cf);            // mov rdi, rcx (1st arg in Win64)
